### ISSUE TITLE
Add an API to merge a model/Modelica code

### ISSUE THREAD

**Issue by adeas31** [OpenModelica/OpenModelica#11553]
### Description

Right now OMEdit uses its internal structures for copy/paste functionality. This causes a problem when user copies something from model A and unloads the model A. Now its not possible to paste the copied data because it has been removed from the clipboard when the model was unloaded.

One solution for this is that OMEdit generates a valid Modelica code when copying the stuff e.g.,

```Modelica
Modelica.Electrical.Analog.Basic.Inductor L annotation(Placement(transformation(origin={-75,38},extent={{-25,-25},{25,25}},rotation=270)));
Modelica.Electrical.Analog.Basic.Resistor Ro annotation(Placement(transformation(origin={-75,-17},extent={{-25,-25},{25,25}},rotation=270)));
Modelica.Electrical.Analog.Basic.Conductor G annotation(Placement(transformation(extent={{-25,38},{25,88}})));
Modelica.Electrical.Analog.Basic.Capacitor C1 annotation(Placement(transformation(origin={25,3},extent={{-25,-25},{25,25}},rotation=270)));
Modelica.Electrical.Analog.Basic.Capacitor C2 annotation(Placement(transformation(origin={-25,3},extent={{-25,-25},{25,25}},rotation=270)));
Modelica.Electrical.Analog.Examples.Utilities.NonlinearResistor Nr annotation(Placement(transformation(origin={75,3},extent={{-25,-25},{25,25}},rotation=270)));
Modelica.Electrical.Analog.Basic.Ground Gnd annotation(Placement(transformation(extent={{-25,-112},{25,-62}})));

equation
connect(L.n, Ro.p) annotation Line(points={{-75,13},{-75,8}});
connect(C2.p, G.p) annotation Line(points={{-25,28},{-25,45.5},{-25,45.5},{-25,63}},color={0,0,255});
connect(L.p, G.p) annotation Line(points={{-75,63},{-25,63}},color={0,0,255});
connect(G.n, Nr.p) annotation Line(points={{25,63},{75,63},{75,28}},color={0,0,255});
connect(C1.p, G.n) annotation Line(points={{25,28},{25,45.5},{25,45.5},{25,63}},color={0,0,255});
connect(Ro.n, Gnd.p) annotation Line(points={{-75,-42},{-75,-62},{0,-62}},color={0,0,255});
connect(C2.n, Gnd.p) annotation Line(points={{-25,-22},{-24,-22},{-24,-62},{0,-62}},color={0,0,255});
connect(Gnd.p, C1.n) annotation Line(points={{0,-62},{25,-62},{25,-22}},color={0,0,255});
connect(Gnd.p, Nr.n) annotation Line(points={{0,-62},{75,-62},{75,-22}},color={0,0,255});

annotation (Text(textColor={0,0,255},extent={{-98,104},{-32,72}},textString="Chua Circuit"));
```

And then passes it to the compiler so the compiler can merge it with the model where it is pasted.

With this solution we can keep the clipboard data since it will not depend on the OMEdit internal structures.

**Comment by adeas31** [comment]
This is needed for #11540

**Comment by casella** [comment]
@adeas31 Does this mean that all formatting is lost when you copy and paste, right?

That would not be a good idea,  is probably not a good idea. People put significant effort in formatting their Modelica code for improved clarity towards human readers. The typical pattern for beginners is to make copies of established models and then play around with them. If the formatting is lost, this doesn't end up well.

**Comment by adeas31** [comment]
This is about copy and paste of elements from one model to another. It is almost impossible to preserve the formatting in that case unless we use the concrete syntax tree that allows us to directly link the element to the Modelica code. If you want to preserve the formatting then just copy and paste the text from one model to another.

**Comment by perost** [comment]
I added a new API `loadClassContentString` in #11574 that hopefully does what you want. When we discussed it I said that it might be best if OMEdit puts the contents in a dummy class, but it ended up making more sense to just pass the contents as in your example without a surrounding class.

The API tries to be a bit intelligent with how it merges things, trying to merge sections without mixing old and new elements too much. It might be a bit overengineered though, let me know what you think.
